File: src/main/java/seremis/geninfusion/soul/allele/AlleleFloat.java
package seremis.geninfusion.soul.allele;

import net.minecraft.nbt.NBTTagCompound;
import seremis.geninfusion.api.soul.EnumAlleleType;
import seremis.geninfusion.soul.Allele;

public class AlleleFloat extends Allele {

    public float value;

    public AlleleFloat(boolean isDominant, float value) {
        super(isDominant, EnumAlleleType.FLOAT);
        this.value = value;
    }

    public AlleleFloat(Object... args) {
        super(args);
        value = (Float) args[1];
        type = EnumAlleleType.FLOAT;
    }

    public AlleleFloat(NBTTagCompound compound) {
        super(compound);
        readFromNBT(compound);
    }

    @Override
    public void writeToNBT(NBTTagCompound compound) {
        super.writeToNBT(compound);
        compound.setFloat("value", value);
    }

    @Override
    public void readFromNBT(NBTTagCompound compound) {
        super.readFromNBT(compound);
        value = compound.getFloat("value");
    }
}

File: src/main/java/seremis/geninfusion/soul/allele/AlleleFloat.scala
package seremis.geninfusion.soul.allele

import net.minecraft.nbt.NBTTagCompound
import seremis.geninfusion.api.soul.EnumAlleleType
import seremis.geninfusion.soul.Allele

class AlleleFloat(isDominant: Boolean, var value: Float) extends Allele(isDominant, EnumAlleleType.FLOAT) {

    def this(args: AnyRef*) {
        this(args(0).asInstanceOf[Boolean], args(1).asInstanceOf[Float])
    }

    def this(compound: NBTTagCompound) {
        this(false, 0.0F)
        readFromNBT(compound)
    }

    override def writeToNBT(compound: NBTTagCompound) {
        super.writeToNBT(compound)
        compound.setFloat("value", value)
    }

    override def readFromNBT(compound: NBTTagCompound) {
        super.readFromNBT(compound)
        value = compound.getFloat("value")
    }
}
